tmc-langs-r/src/test/java/fi/helsinki/cs/tmc/langs/r/RPluginTest.java

```diff
@@ -0,0 +1,203 @@
+    public void runTestsRunResultAsExpectedSimpleAllPass() {
+        RunResult result = plugin.runTests(simpleAllTestsPassProject);
+
+        assertEquals(RunResult.Status.PASSED, result.status);
+
+        ImmutableList<TestResult> results = result.testResults;
+        testResultAsExpected(results.get(0), true,
+                "ret_true works.", new String[]{"r1", "r1.1"});
+        testResultAsExpected(results.get(1), true,
+                "ret_one works.", new String[]{"r1", "r1.2"});
+        testResultAsExpected(results.get(2), true,
+                "add works.", new String[]{"r1", "r1.3", "r1.4"});
+        testResultAsExpected(results.get(3), true,
+                "minus works", new String[]{"r2", "r2.1"});
+    }
+
+    @Test
+    public void runTestsRunResultAsExpectedSimpleSomeFail() {
+        RunResult result = plugin.runTests(simpleSomeTestsFailProject);
+
+        assertEquals(RunResult.Status.TESTS_FAILED, result.status);
+
+        ImmutableList<TestResult> results = result.testResults;
+        testResultAsExpected(results.get(0), true,
+                "ret_true works.", new String[]{"r1", "r1.1"});
+        testResultAsExpected(results.get(1), true,
+                "ret_one works.", new String[]{"r1", "r1.2"});
+        testResultAsExpected(results.get(2), true,
+                "add works.", new String[]{"r1", "r1.3", "r1.4"});
+        testResultAsExpected(results.get(3), false,
+                "ret_false returns true", new String[]{"r1", "r1.5"});
+        testResultAsExpected(results.get(4), true,
+                "ret_true works but there are no points.", new String[]{"r1"});
+    }
+
+    @Test
+    public void runTestsCreatesRunResultWithCorrectStatusWhenSourceCodeHasError() {
+        RunResult res = plugin.runTests(simpleSourceCodeErrorProject);
+
+        assertEquals(RunResult.Status.COMPILE_FAILED, res.status);
+    }
+
+    @Test
+    public void exerciseIsCorrectTypeIfItContainsRFolder() {
+        Path testCasesRoot = TestUtils.getPath(getClass(), "recognition_test_cases");
+        Path project = testCasesRoot.resolve("R_folder");
+
+        assertTrue(plugin.isExerciseTypeCorrect(project));
+    }
+
+    @Test
+    public void exerciseIsCorrectTypeIfItContainsTestthatFolder() {
+        Path testCasesRoot = TestUtils.getPath(getClass(), "recognition_test_cases");
+        Path project = testCasesRoot.resolve("testthat_folder");
+
+        assertTrue(plugin.isExerciseTypeCorrect(project));
+    }
+
+    @Test
+    public void exerciseIsCorrectTypeIfItContainsTestthatFile() {
+        Path testCasesRoot = TestUtils.getPath(getClass(), "recognition_test_cases");
+        Path project = testCasesRoot.resolve("testthat_folder")
+                                    .resolve("tests");
+
+        File testThatR = new File(project.toAbsolutePath().toString() + "/testthat.R");
+        assertTrue(testThatR.exists());
+    }
+
+
+    @Test
+    public void getStudentFilePolicyReturnsRStudentFilePolicy() {
```

Comment:
What's the value of this test?

Reply:
The python plugin has a similar test so I assumed it was necessary/helpful to include.